src/sentry/rules/filters/issue_occurrences.py

```diff
@@ -26,3 +29,26 @@ def passes(self, event: GroupEvent, state: EventState) -> bool:
         # around this by including pending updates from buffers to improve accuracy.
         issue_occurrences: int = event.group.times_seen_with_pending
         return issue_occurrences >= value
+
+    def passes_activity(self, condition_activity: ConditionActivity) -> bool:
+        try:
+            value = int(self.get_option("value"))
+        except (TypeError, ValueError):
+            return False
+
+        try:
+            group = Group.objects.get_from_cache(id=condition_activity.group_id)
+        except Group.DoesNotExist:
+            return False
+
+        now = timezone.now()
+        if now == group.first_seen:
+            return group.times_seen >= value
+        # assumes uniform distribution of error occurrences between first_seen and now
+        guess = (
+            (condition_activity.timestamp - group.first_seen)
+            / (now - group.first_seen)
+            * group.times_seen
+        )
```

Comment:
would it be too slow to make a tsdb query here?

Reply:
What do you want to query tsdb for?

Reply:
Instead of guessing, we could query the number of events between `group.first_seen` and `condition_activity.timestamp` for the group to get an exact count

Reply:
So this only occurs in the ui when people are making a test rule right? I forget how the other logic works, is this being done for many groups or just a single one?

Reply:
Yep, it only occurs in the create/edit alert rule page. This could be done for each `ConditionActivity` (group creation, regression, reappearance), but we do have a cap `CONDITION_ACTIVITY_LIMIT` on how many activities each condition can return (currently 1k per condition)

Reply:
Right, so this could result in 1k queries to TSDB? That's not something we can support, unfortunately.

If there was a way to batch this it might be ok though. So just say you made a query with 1000 group ids that would be ok. Maybe the query could be something like (psuedosql)

```
select distinct(group_id)
from errors
where group_id in (<group_ids>)
group by group_id, <date_rounded_to_time_bucket>
having count(*) > <value>
```
Not sure if you need the actual distinct group_ids here or just the counts, but either should be possible.

Reply:
Yeah I don't think this is doable by batching, each `ConditionActivity` could even potentially be the same group, and there aren't fixed time buckets, but instead we want for each `ConditionActivity` the time range between `ConditionActivity.timestamp` and `ConditionActivity.group.first_seen`

Reply:
Hmm actually it might be possible, if for each group we can round every event to the next `ConditionActivity` that occurs then we can go over the results after and accumulate the sum to get what we want

Reply:
Well it's kinda complicated and i'm not 100% sure it'll even work, if you think the current method isn't that bad then I think it can go in a separate pr

Reply:
The current method should be ok for now, we can try improve it later if we think it's important